### PR TITLE
go-font: 2017-03-30 -> 2.010

### PR DESCRIPTION
**Pull request by bdd**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/fonts/go-font/default.nix [hunk 1]
```diff
@@ -1,11 +1,11 @@
 { lib, fetchgit }:
 
 let
-  version = "2017-03-30";
+  version = "2022-06-17";
```
Comment (sternenseemann) [inline pkgs/data/fonts/go-font/default.nix:4]: Shouldn't the version be 2.010 now? See https://go.googlesource.com/image/+/41969df76e82aeec85fa3821b1e24955ea993001

Comment (bdd) [reply pkgs/data/fonts/go-font/default.nix:4]: I spent quite a bit of time on deciding what this should be :)
I first changed it to `2.010` as it was declared in the commit. ...But then it felt odd to declare a version where the upstream doesn't use any VCS tags. Considered maybe it should be `2.010-2022-06-17` but it was long for no good reason. Sighed and fell back to datestamp.

If you as the maintainer are OK with `2.010` directly to a commit hash, I'll happily update the PR.

Comment (sternenseemann) [reply pkgs/data/fonts/go-font/default.nix:4]: I'm in favor! I think you forgot to update the version string in the expression, though.

Comment (bdd) [reply pkgs/data/fonts/go-font/default.nix]: git amend alias mistake. fixed.

**Pull request merged**